File: backend/src/BiteRight.Application/Dtos/Units/UnitDto.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: UnitDto.cs
// # Author: Łukasz Sobczak
// # Created: 12-02-2024
// # ==============================================================================

#region

using System;
using BiteRight.Domain.Languages;
using BiteRight.Domain.Units;

#endregion

namespace BiteRight.Application.Dtos.Units;

public class UnitDto
{
    public UnitDto(
        Guid id,
        string name,
        string abbreviation,
        UnitSystemDto unitSystem
    )
    {
        Id = id;
        Name = name;
        Abbreviation = abbreviation;
        UnitSystem = unitSystem;
    }

    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Abbreviation { get; set; }
    public UnitSystemDto UnitSystem { get; set; }

    public static UnitDto FromDomain(
        Unit unit,
        LanguageId languageId
    )
    {
        return new UnitDto(
            unit.Id,
            unit.GetName(languageId),
            unit.GetAbbreviation(languageId),
            (UnitSystemDto)unit.UnitSystem
        );
    }
}

File: mobile/app/src/main/java/com/sobczal2/biteright/dto/units/UnitDto.kt
package com.sobczal2.biteright.dto.units

import java.util.UUID

data class UnitDto(
    val id: UUID,
    val name: String,
    val abbreviation: String,
    val unitSystem: UnitSystemDto
) {
    companion object {
        val Empty = UnitDto(
            id = UUID.randomUUID(),
            name = "",
            abbreviation = "",
            unitSystem = UnitSystemDto.Empty
        )
    }
}